src/dune_pkg/sys_poll.ml

```diff
@@ -96,9 +96,9 @@ let os ~path =
 ;;
 
 let maybe_read_lines p =
-  match Io.String_path.lines_of_file p with
-  | s -> Some s
-  | exception Unix.Unix_error (Unix.ENOENT, _, _) -> None
+  if Sys.file_exists p && Sys.is_regular_file p
+  then Some (Io.String_path.lines_of_file p)
+  else None
```

Comment:
Wouldn't it be better to catch the `Sys_error` and then do the checks? That way the `stat` calls only happen on the exception case and don't need to be paid on every invocation of the function.

Reply:
Good point. I've changed it to optimistically read lines and to return `None` if `Sys_error` is raised. I'm not sure if I should keep the checks (`Sys.file_exists p && Sys.is_regular_file p`) in the exception case or just return `None` in response to all `Sys_error` exceptions.

Reply:
`Sys.is_regular_file` only exists starting with 5.1, you'd need to use `stat` instead, but for now I'd just leave it out until it proves to be a problem.